dev/tasks/python-wheels/github.linux.yml

```diff
@@ -22,15 +22,37 @@
 jobs:
   build:
     name: "Build wheel for manylinux {{ manylinux_version }}"
+    {% if arch == "amd64" %}
     runs-on: ubuntu-latest
+    {% else %}
+    runs-on: ["self-hosted", "Linux", "arm64"]
+    {% endif %}
     env:
       # archery uses these environment variables
+      {% if arch == "amd64" %}
       ARCH: amd64
+      {% else %}
+      ARCH: arm64v8
+      {% endif %}
       PYTHON: "{{ python_version }}"
 
     steps:
       {{ macros.github_checkout_arrow()|indent }}
+      {% if arch == "amd64" %}
       {{ macros.github_install_archery()|indent }}
+      {% else %}
+      - name: add path
+        run: echo "$HOME/.local/bin" >>"$GITHUB_PATH"
+      - name: Install python # Workaround for arm64 https://github.com/actions/setup-python/issues/108
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y python3-pip
+          pip3 install -U pip
+      - name: Install Archery
+        shell: bash
+        run: pip3 install -e arrow/dev/archery[all]
```

Comment:
Could you do this in `macros.jinja` like https://github.com/apache/arrow/pull/34717/files#diff-77e693ebc06aba9fdd8c4b8d27e8ddcf54d91db8dbb7c060910e4a9c25d76a3c ?